src/main/java/org/cryptomator/ui/health/CheckExecutor.java

```diff
@@ -0,0 +1,113 @@
+package org.cryptomator.ui.health;
+
+import org.cryptomator.common.vaults.Vault;
+import org.cryptomator.cryptofs.VaultConfig;
+import org.cryptomator.cryptofs.health.api.DiagnosticResult;
+import org.cryptomator.cryptolib.api.CryptorProvider;
+import org.cryptomator.cryptolib.api.Masterkey;
+
+import javax.inject.Inject;
+import javafx.concurrent.Task;
+import java.nio.file.Path;
+import java.security.SecureRandom;
+import java.util.ArrayDeque;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicReference;
+
+@HealthCheckScoped
+public class CheckExecutor {
+
+	private final Path vaultPath;
+	private final SecureRandom csprng;
+	private final Masterkey masterkey;
+	private final VaultConfig vaultConfig;
+	private final ExecutorService sequentialExecutor;
+	private final Queue<CheckTask> runningTasks;
```

Comment:
I believe this should be a thread-safe collection (like `BlockingQueue`) and the task should "remove itself" when finished/failed.

Reply:
(this behaviour will be for free, if when applying my next suggestion)

Reply:
Fixed in b4a530b